settings.py

```diff
@@ -0,0 +1,221 @@
+
+import chess
+import chess.pgn
+import psutil
+
+
+class Order(Enum):
+    LONG_TO_SHORT = "Long lines to short lines"
+    SHORT_TO_LONG = "Short lines to long lines"
+
+
+class Book:
+    def __init__(self, name: str, pgn: str):
+        self.name = name
+        self.pgn = pgn
+
+    def __str__(self) -> str:
+        return f"Name: {self.name}\nPGN: {self.pgn}"
+
+    def is_valid_pgn(self) -> bool:
+        if not self.pgn:
+            return False
+
+        # poor man's validation: if chess.pgn throws when parsing a PGN, the PGN is not valid
+        game = chess.pgn.read_game(io.StringIO(self.pgn))
+        return len(game.errors) == 0
+
+
+class BookSettings:
+    def __init__(self) -> None:
+        self.order: Order = Order.SHORT_TO_LONG
+        self.books_string: str = "Book A\n1. e4 e5\n\nBook B\n1. e4 e5 2. f4"
+
+    def order_callback(self, _, order_value):
+        self.order = Order(order_value)
+
+    def books_string_callback(self, _, books_string):
+        self.books_string = books_string
+
+    def get_books(self) -> List[Book]:
+        books = list()
+        lines = self.books_string.splitlines()
+        non_empty_lines = iter([line.strip() for line in lines if line and line.strip()])
+        for book_name in non_empty_lines:
+            books.append(Book(book_name, next(non_empty_lines, None)))
+        return books
+
+
+class Variant(Enum):
+    STANDARD = "standard"
+    CHESS960 = "chess960"
+    CRAZYHOUSE = "crazyhouse"
+    ANTICHESS = "antichess"
+    ATOMIC = "atomic"
+    HORDE = "horde"
+    KING_OF_THE_HILL = "kingOfTheHill"
+    RACING_KINGS = "racingKings"
+    THREE_CHECK = "threeCheck"
+
+
+class Speed(Enum):
+    ULTRA_BULLET = "ultraBullet"
+    BULLET = "bullet"
+    BLITZ = "blitz"
+    RAPID = "rapid"
+    CLASSICAL = "classical"
+    CORRESPONDENCE = "correspondence"
+
+
+class Rating(Enum):
+    ELO_1600 = "1600"
+    ELO_1800 = "1800"
+    ELO_2000 = "2000"
+    ELO_2200 = "2200"
+    ELO_2500 = "2500"
+
+
+class DatabaseSettings:
+    def __init__(self):
+        self.variant: Variant = Variant.STANDARD
+        self.speeds: List[Speed] = [Speed.RAPID, Speed.CLASSICAL]
+        self.ratings: List[Rating] = [Rating.ELO_1800, Rating.ELO_2000, Rating.ELO_2200]
+        self.moves: int = 10
+
+    def variant_callback(self, _, variant_name):
+        self.variant = Variant[variant_name]
+
+    def speed_callback(self, _, is_selected, speed_name):
+        speed = Speed[speed_name]
+        if is_selected:
+            self.speeds.append(speed)
+        else:
+            self.speeds.remove(speed)
+
+    def rating_callback(self, _, is_selected, rating_name):
+        rating = Rating[rating_name]
+        if is_selected:
+            self.ratings.append(rating)
+        else:
+            self.ratings.remove(rating)
+
+    def moves_callback(self, _, moves):
+        if moves > 5:
+            self.moves = moves
+
+
+class MoveSelectionSettings:
+    def __init__(self):
```

Comment:
When someone adds a new setting, they'll have to:
- add it into one of those `__init__` methods
- create a callback like `depth_callback` below - it's a method that's called by the GUI after a user interacts with a widget
- add a new UI widget in `gui.py` (probably by copy-pasting one of the existing ones and adjusting some values)
-* **important** - make sure that the callback of the new widget is calling the newly created callback for the new setting, not the copied callback - otherwise the wrong setting will be updated